nncf/quantization/quantizer_propagation.py

```diff
@@ -590,28 +735,35 @@ def traverse_graph(self, curr_node_key: str,
                        traverse_function: Callable[[str, Any], Tuple[bool, Any]],
                        output: Any,
                        traverse_forward: bool = True) -> Any:
-        return self._traverse_graph_recursive_helper(curr_node_key, traverse_function, output, traverse_forward)
+        visited_node_keys = set()  # type: Set[str]
+        return self._traverse_graph_recursive_helper(curr_node_key,
+                                                     visited_node_keys,
+                                                     traverse_function, output, traverse_forward)
 
-    def _traverse_graph_recursive_helper(self, curr_node_key: str,
+    def _traverse_graph_recursive_helper(self, curr_node_key: str, visited_node_keys: Set[str],
                                          traverse_function: Callable[[str, Any], Tuple[bool, Any]],
                                          output: Any, traverse_forward: bool):
         is_finished, output = traverse_function(curr_node_key, output)
+        visited_node_keys.add(curr_node_key)
         node_keys_holder = self.succ if traverse_forward else self.pred
         if not is_finished:
             for node_key in node_keys_holder[curr_node_key]:
-                self._traverse_graph_recursive_helper(node_key, traverse_function, output, traverse_forward)
+                if node_key not in visited_node_keys:
+                    self._traverse_graph_recursive_helper(node_key, visited_node_keys,
+                                                          traverse_function, output, traverse_forward)
         return output
 
     def _get_next_prop_quantizer_id(self):
         self._created_prop_quantizer_counter += 1
         return self._created_prop_quantizer_counter
 
     def _is_position_accepting(self, ip_node_key: str):
-        node = self.nodes[ip_node_key]
-        insertion_type = node[QuantizerPropagationStateGraph.INSERTION_POINT_DATA_NODE_ATTR].insertion_type
-        if insertion_type == InsertionType.OPERATOR_POST_HOOK:
-            return True
-        return False
+        # node = self.nodes[ip_node_key]
+        # insertion_type = node[QuantizerPropagationStateGraph.INSERTION_POINT_DATA_NODE_ATTR].insertion_type
+        # if insertion_type == InsertionType.OPERATOR_POST_HOOK:
+        #     return True
+        # return False
+        return True
```

Comment:
Now that all insertion points, both pre- and post-hook, are functional, this function is mostly obsolete.